imap_processing/tests/spice/test_geometry.py

```diff
@@ -23,3 +29,47 @@ def test_imap_state(et, use_test_metakernel):
         np.testing.assert_array_equal(state.shape, (len(et), 6))
     else:
         assert state.shape == (6,)
+
+
+def test_get_spacecraft_spin_phase():
+    """Test get_spacecraft_spin_phase() with generated spin data."""
+
+    start_time = 453051323.0
+
+    spin_phases = get_spacecraft_spin_phase(query_met_times=start_time)
+
+    # Uncomment this once spin phase calculation is implemented
+    # assert np.all(
+    #     (spin_phases >= 0) & (spin_phases <= 1)
+    # ), "Spin phases must be in [0, 1] range."
+
+    # Ensure the length of spin phases matches the query times
+    assert len(spin_phases) == 1, "Spin phases length should match query times length."
+
+
+def test_get_spin_data(generate_spin_data, tmpdir):
+    """Test get_spin_data() with generated spin data."""
+
+    # SWE test data time minus 56120 seconds to get mid-night time
+    start_time = 453051323.0 - 56120
+    spin_df = generate_spin_data(start_time)
+    spin_csv_file_path = Path(tmpdir) / "spin_data.spin.csv"
```

Comment:
I think tmpdir is already a Path?
```suggestion
    spin_csv_file_path = tmpdir / "spin_data.spin.csv"
```

Reply:
Without this, the call `.suffix` did not work but since I removed that from your above feedback, I can revert this back.